hummingbot/connector/exchange/xrpl/xrpl_exchange.py

```diff
@@ -0,0 +1,1430 @@
+        custom_markets: Dict[str, XRPLMarket] = None,
+    ):
+        self._xrpl_secret_key = xrpl_secret_key
+        self._wss_node_url = wss_node_url
+        self._wss_second_node_url = wss_second_node_url
+        self._xrpl_client = AsyncWebsocketClient(self._wss_node_url)
+        self._xrpl_place_order_client = AsyncWebsocketClient(self._wss_second_node_url)
+        self._trading_required = trading_required
+        self._trading_pairs = trading_pairs
+        self._auth: XRPLAuth = self.authenticator
+        self._trading_pair_symbol_map: Optional[Mapping[str, str]] = None
+        self._trading_pair_fee_rules: Dict[str, Dict[str, Any]] = {}
+        self._xrpl_client_lock = asyncio.Lock()
+        self._xrpl_place_order_client_lock = asyncio.Lock()
+        self._nonce_creator = NonceCreator.for_microseconds()
+        self._custom_markets = custom_markets or {}
+
+        super().__init__(client_config_map)
+
+    @staticmethod
+    def xrpl_order_type(order_type: OrderType) -> str:
+        return CONSTANTS.XRPL_ORDER_TYPE[order_type]
+
+    @staticmethod
+    def to_hb_order_type(order_type: str) -> OrderType:
+        return OrderType[order_type]
+
+    @property
+    def authenticator(self) -> XRPLAuth:
+        return XRPLAuth(xrpl_secret_key=self._xrpl_secret_key)
+
+    @property
+    def name(self) -> str:
+        return CONSTANTS.EXCHANGE_NAME
+
+    @property
+    def rate_limits_rules(self):
+        return CONSTANTS.RATE_LIMITS
+
+    @property
+    def domain(self):
+        return "Not Supported"
+
+    @property
+    def client_order_id_max_length(self):
+        return CONSTANTS.MAX_ORDER_ID_LEN
+
+    @property
+    def client_order_id_prefix(self):
+        return CONSTANTS.HBOT_ORDER_ID_PREFIX
+
+    @property
+    def trading_rules_request_path(self):
+        return ""
+
+    @property
+    def trading_pairs_request_path(self):
+        return ""
+
+    @property
+    def check_network_request_path(self):
+        return ""
+
+    @property
+    def trading_pairs(self):
+        return self._trading_pairs
+
+    @property
+    def is_cancel_request_in_exchange_synchronous(self) -> bool:
+        return False
+
+    @property
+    def is_trading_required(self) -> bool:
+        return self._trading_required
+
+    @property
+    def node_url(self) -> str:
+        return self._wss_node_url
+
+    @property
+    def second_node_url(self) -> str:
+        return self._wss_second_node_url
+
+    @property
+    def auth(self) -> XRPLAuth:
+        return self._auth
+
+    def supported_order_types(self):
+        return [OrderType.LIMIT, OrderType.LIMIT_MAKER, OrderType.MARKET]
+
+    def _is_request_exception_related_to_time_synchronizer(self, request_exception: Exception):
+        # API documentation does not clarify the error message for timestamp related problems
+        return False
+
+    def _is_order_not_found_during_status_update_error(self, status_update_exception: Exception) -> bool:
```

Comment:
this will be important to implement in case that we request an order that is in memory but the update of it wasn't correct

Reply:
Thanks, added a TODO for this